gridstatus/ieso.py

```diff
@@ -0,0 +1,473 @@
+                ".//DeliveryDate",
+                NAMESPACES_FOR_XML,
+            ).text
+
+            for zonal_demand in zonal_demands.findall(
+                ".//ZonalDemand/*",
+                NAMESPACES_FOR_XML,
+            ):
+                # The zone name is the tag name without the namespace
+                zone_name = zonal_demand.tag[(zonal_demand.tag.rfind("}") + 1) :]
+
+                for demand in zonal_demand.findall(".//Demand", NAMESPACES_FOR_XML):
+                    hour = demand.find(".//DeliveryHour", NAMESPACES_FOR_XML).text
+                    energy_mw = demand.find(".//EnergyMW", NAMESPACES_FOR_XML).text
+
+                    data.append(
+                        {
+                            "DeliveryDate": delivery_date,
+                            "Zone": zone_name,
+                            "DeliveryHour": hour,
+                            "EnergyMW": energy_mw,
+                        },
+                    )
+
+        df = pd.DataFrame(data)
+
+        # Convert columns to appropriate data types
+        df["DeliveryHour"] = df["DeliveryHour"].astype(int)
+        df["EnergyMW"] = df["EnergyMW"].astype(float)
+        df["DeliveryDate"] = pd.to_datetime(df["DeliveryDate"])
+
+        df["Interval Start"] = (
+            df["DeliveryDate"] + pd.to_timedelta(df["DeliveryHour"], unit="h")
+        ).dt.tz_localize(self.default_timezone)
+
+        df["Interval End"] = df["Interval Start"] + pd.Timedelta(hours=HOUR_INTERVAL)
+
+        # Pivot the table to wide
+        pivot_df = df.pivot_table(
+            index=["Interval Start", "Interval End"],
+            columns="Zone",
+            values="EnergyMW",
+            aggfunc="first",
+        ).reset_index()
+
+        pivot_df["Published Time"] = pd.Timestamp(
+            published_time,
+            tz=self.default_timezone,
+        )
+
+        pivot_df = utils.move_cols_to_front(
+            pivot_df,
+            [
+                "Interval Start",
+                "Interval End",
+                "Published Time",
+                "Ontario",
+            ],
+        )
+
+        pivot_df.columns.name = None
+
+        col_mapper = {
+            col: f"{col} Load Forecast" for col in ["Ontario", "East", "West"]
+        }
+
+        pivot_df = pivot_df.rename(columns=col_mapper)
+
+        # If no end is provided, return data from single date
+        if not end:
+            return pivot_df[
+                pivot_df["Interval Start"].dt.date == date_only
+            ].reset_index(drop=True)
+
+        end_date_only = pd.to_datetime(end).date()
+
+        return pivot_df[
+            (pivot_df["Interval Start"].dt.date >= date_only)
+            & (pivot_df["Interval End"].dt.date <= end_date_only)
+        ].reset_index(drop=True)
+
+    # TODO add fuel mix. http://reports.ieso.ca/public/GenOutputbyFuelHourly/
+    def get_fuel_mix(self, date, end=None, verbose=False):
+        pass
+
+    def _today(self):
```

Comment:
you can use `utils._handle_date("today", tz=tz)` rather than define this helper

https://github.com/kmax12/gridstatus/blob/main/gridstatus/utils.py#L99C11-L99C11